src/ol-ext/format/featurehash.js

```diff
@@ -0,0 +1,940 @@
+  for (var i = 0, ii = polygons.length; i < ii; ++i) {
+    var rings = polygons[i].split('\'');
+    var ends = endss[i] = [];
+    for (var j = 0, jj = rings.length; j < jj; ++j) {
+      flatCoordinates = this.decodeCoordinates_(rings[j], flatCoordinates);
+      var end = flatCoordinates.length;
+      if (j === 0) {
+        flatCoordinates[end++] = flatCoordinates[0];
+        flatCoordinates[end++] = flatCoordinates[1];
+      } else {
+        flatCoordinates[end++] = flatCoordinates[ends[j - 1]];
+        flatCoordinates[end++] = flatCoordinates[ends[j - 1] + 1];
+      }
+      ends[j] = end;
+    }
+  }
+  var multipolygon = new ol.geom.MultiPolygon(null);
+  multipolygon.setFlatCoordinates(
+      ol.geom.GeometryLayout.XY, flatCoordinates, endss);
+  return multipolygon;
+};
+
+
+/**
+ * @param {string} text Text.
+ * @param {ol.Feature} feature Feature.
+ * @private
+ */
+ngeo.format.FeatureHash.setStyleInFeature_ = function(text, feature) {
+  var fillColor, fontSize, fontColor, pointRadius, strokeColor, strokeWidth;
+  var parts = text.split('\'');
+  for (var i = 0; i < parts.length; ++i) {
+    var part = decodeURIComponent(parts[i]);
+    var keyVal = part.split('*');
+    goog.asserts.assert(keyVal.length === 2);
+    var key = keyVal[0];
+    var val = keyVal[1];
+    if (key === 'fillColor') {
+      fillColor = val;
+    } else if (key == 'fontSize') {
+      fontSize = val;
+    } else if (key == 'fontColor') {
+      fontColor = val;
+    } else if (key == 'pointRadius') {
+      pointRadius = +val;
+    } else if (key == 'strokeColor') {
+      strokeColor = val;
+    } else if (key == 'strokeWidth') {
+      strokeWidth = +val;
+    }
+  }
+  var fillStyle = null;
+  if (goog.isDef(fillColor)) {
+    fillStyle = new ol.style.Fill({
+      color: fillColor
+    });
+  }
+  var strokeStyle = null;
+  if (goog.isDef(strokeColor) && goog.isDef(strokeWidth)) {
+    strokeStyle = new ol.style.Stroke({
+      color: strokeColor,
+      width: strokeWidth
+    });
+  }
+  var imageStyle = null;
+  if (goog.isDef(pointRadius)) {
+    imageStyle = new ol.style.Circle({
+      radius: pointRadius,
+      fill: fillStyle,
+      stroke: strokeStyle
+    });
+    fillStyle = strokeStyle = null;
+  }
+  var textStyle = null;
+  if (goog.isDef(fontSize) && goog.isDef(fontColor)) {
+    textStyle = new ol.style.Text({
+      font: fontSize + ' sans-serif',
+      fill: new ol.style.Fill({
+        color: fontColor
+      })
+    });
+  }
+  var style = new ol.style.Style({
+    fill: fillStyle,
+    image: imageStyle,
+    stroke: strokeStyle,
+    text: textStyle
+  });
+  feature.setStyle(style);
+};
+
+
+/**
+ * @param {ol.geom.Geometry} geometry Geometry.
+ * @param {olx.format.WriteOptions=} opt_options Write options.
+ * @return {string} Encoded geometry.
+ * @this {ngeo.format.FeatureHash}
+ * @private
+ */
+ngeo.format.FeatureHash.writeLineStringGeometry_ =
+    function(geometry, opt_options) {
```

Comment:
Yes. There are places where it cannot be removed though, because it's part of the signature from the parent method.